docs/proposals/cascading-deletion.md

```diff
@@ -0,0 +1,247 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Overview
+
+Currently most cascading deletion logic is implemented at client-side. For example, when deleting a replica set, kubectl uses a reaper to delete the created pods and then delete the replica set. We plan to move the cascading deletion to the server to simplify the client-side logic. In this proposal, we present two alternative designs. [Design #1](#design-1-tombstone--gc) uses the tombstone mechanism, and [Desig #2](#design-2-finalizer-framework--gc) uses the finalizer mechanism. Some
+common design choices faced by both designs are described in [common designs](#common-designs).
+
+# Goals
+
+* Supporting cascading deletion at the server-side.
```

Comment:
No, I don't think we should require that. It should work if the user
chooses to use it, though.

On Mon, Apr 11, 2016 at 11:58 AM, Chao Xu <notifications@github.com> wrote:

> In docs/proposals/cascading-deletion.md
> <https://github.com/kubernetes/kubernetes/pull/23656#discussion_r59261138>
> :
>
> > +[releases.k8s.io](http://releases.k8s.io).
> > +</strong>
> > +--
> > +
> > +<!-- END STRIP_FOR_RELEASE -->
> > +
> > +<!-- END MUNGE: UNVERSIONED_WARNING -->
> > +
> > +# Overview
> > +
> > +Currently most cascading deletion logic is implemented at client-side. For example, when deleting a replica set, kubectl uses a reaper to delete the created pods and then delete the replica set. We plan to move the cascading deletion to the server to simplify the client-side logic. In this proposal, we present two alternative designs. [Design #1](#design-1-tombstone--gc) uses the tombstone mechanism, and [Desig #2](#design-2-finalizer-framework--gc) uses the finalizer mechanism. Some
> > +common design choices faced by both designs are described in [common designs](#common-designs).
> > +
> > +# Goals
> > +
> > +* Supporting cascading deletion at the server-side.
>
> In the HPA case, I imagine when the HPA is created, there should be a
> validation rule to check if the ParentReferences is a super set of
> ScaleRef, does it make sense?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23656/files/59289b26e577ada2800485b30e8440d0bac0cab2#r59261138>
>